dev/NavigationView/NavigationView.cpp

```diff
@@ -3999,6 +4000,7 @@ void NavigationView::OnPropertyChanged(const winrt::DependencyPropertyChangedEve
     else if (property == s_PaneFooterProperty)
     {
         UpdatePaneLayout();
+        AddNavigationViewReferenceToPaneFooter();
```

Comment:
args.NewValue has the element set as the PaneFooter, so you can just check if that is a NavViewItemBase instead of walking the tree.